server/handlers/logic.go

```diff
@@ -0,0 +1,89 @@
+package handlers
+
+import (
+	"io"
+
+	"github.com/Sirupsen/logrus"
+	ctxu "github.com/docker/distribution/context"
+	"golang.org/x/net/context"
+
+	"github.com/docker/notary/server/errors"
+	"github.com/docker/notary/server/snapshot"
+	"github.com/docker/notary/server/storage"
+	"github.com/docker/notary/server/timestamp"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/signed"
+)
+
+func getRole(ctx context.Context, logger ctxu.Logger, w io.Writer, store storage.MetaStore, gun, role, checksum string) error {
+	var (
+		out []byte
+		err error
+	)
+	if checksum == "" {
+		// the timestamp and snapshot might be server signed so are
+		// handled specially
+		switch role {
+		case data.CanonicalTimestampRole, data.CanonicalSnapshotRole:
+			return getMaybeServerSigned(ctx, w, logger, store, gun, role)
```

Comment:
why not follow the same pattern and have this be `out, err = getMaybeServerSigned(...)`?